### ISSUE TITLE
SERVO_TILT / Servo gimbal doesn't work on SPRacingF3 (release 1.5).

### ISSUE THREAD

**Issue by Xe54** [iNavFlight/inav#1014]
I can't get my gimbal working on SPRF3 acro board. Controls for pitch and roll axis are connected to the motor outputs 7 and 8 respectively. The way it is working with the current version of CleanFlight (1.14.2) is to enable SERVO_TILT in configurations and to set Servo 0 and Servo 1 to desired receiver channels (A2 and A3 in my case). Done. But it doesn't work that way in iNav. I would like to know if it's just a changed handling/channel mapping or is it a real bug? I also tried it with CHANNEL_FORWARDING and it doesn't work either.

**Comment by oleost** [comment]
Hi. Could you provide more info:

Your current setup works if you flash cleanflight?
Can you test older iNav versions and check if it worked on any previous versions? If so find out where it stopped working.

Provide both **bootlog** and **dump** output from CLI here.

**Comment by Xe54** [comment]
Hi, I know this issue since iNav version 1.2.0, but it could be older. I could perform some more tests with older versions this weekend if necessary. My setup worked fine with each version of Cleanflight i've tried before (since 1.13.0). Here is my current dump:

# dump

# version
# INAV/SPRACINGF3 1.5.0 Dec 27 2016 / 10:12:30 (bfa1661)
# pflags
# Persistent config flags: 0x00000000
# dump master

# mixer
mixer QUADX
mmix reset
smix reset


# feature
feature -RX_PPM
feature -VBAT
feature -UNUSED_1
feature -RX_SERIAL
feature -MOTOR_STOP
feature -SERVO_TILT
feature -SOFTSERIAL
feature -GPS
feature -FAILSAFE
feature -SONAR
feature -TELEMETRY
feature -CURRENT_METER
feature -3D
feature -RX_PARALLEL_PWM
feature -RX_MSP
feature -RSSI_ADC
feature -LED_STRIP
feature -DASHBOARD
feature -UNUSED_2
feature -BLACKBOX
feature -CHANNEL_FORWARDING
feature -TRANSPONDER
feature -AIRMODE
feature -SUPEREXPO
feature -VTX
feature -RX_SPI
feature -SOFTSPI
feature -PWM_SERVO_DRIVER
feature -PWM_OUTPUT_ENABLE
feature -OSD
feature VBAT
feature RX_SERIAL
feature MOTOR_STOP
feature SERVO_TILT
feature SOFTSERIAL
feature GPS
feature FAILSAFE
feature TELEMETRY
feature CURRENT_METER
feature BLACKBOX
feature PWM_OUTPUT_ENABLE


# beeper
beeper GYRO_CALIBRATED
beeper HW_FAILURE
beeper RX_LOST
beeper RX_LOST_LANDING
beeper DISARMING
beeper ARMING
beeper ARMING_GPS_FIX
beeper BAT_CRIT_LOW
beeper BAT_LOW
beeper GPS_STATUS
beeper RX_SET
beeper ACC_CALIBRATION
beeper ACC_CALIBRATION_FAIL
beeper READY_BEEP
beeper MULTI_BEEPS
beeper DISARM_REPEAT
beeper ARMED
beeper SYSTEM_INIT
beeper ON_USB
beeper LAUNCH_MODE


# map
map TAER1234


# serial
serial 0 64 115200 38400 0 115200
serial 1 2 115200 115200 0 115200
serial 2 32 115200 38400 0 115200
serial 30 1 57600 38400 0 115200
serial 31 256 115200 38400 0 115200


# led
led 0 0,0::C:0
led 1 0,0::C:0
led 2 0,0::C:0
led 3 0,0::C:0
led 4 0,0::C:0
led 5 0,0::C:0
led 6 0,0::C:0
led 7 0,0::C:0
led 8 0,0::C:0
led 9 0,0::C:0
led 10 0,0::C:0
led 11 0,0::C:0
led 12 0,0::C:0
led 13 0,0::C:0
led 14 0,0::C:0
led 15 0,0::C:0
led 16 0,0::C:0
led 17 0,0::C:0
led 18 0,0::C:0
led 19 0,0::C:0
led 20 0,0::C:0
led 21 0,0::C:0
led 22 0,0::C:0
led 23 0,0::C:0
led 24 0,0::C:0
led 25 0,0::C:0
led 26 0,0::C:0
led 27 0,0::C:0
led 28 0,0::C:0
led 29 0,0::C:0
led 30 0,0::C:0
led 31 0,0::C:0


# color
color 0 0,0,0
color 1 0,255,255
color 2 0,0,255
color 3 30,0,255
color 4 60,0,255
color 5 90,0,255
color 6 120,0,255
color 7 150,0,255
color 8 180,0,255
color 9 210,0,255
color 10 240,0,255
color 11 270,0,255
color 12 300,0,255
color 13 330,0,255
color 14 0,0,0
color 15 0,0,0


# mode_color
mode_color 0 0 1
mode_color 0 1 11
mode_color 0 2 2
mode_color 0 3 13
mode_color 0 4 10
mode_color 0 5 3
mode_color 1 0 5
mode_color 1 1 11
mode_color 1 2 3
mode_color 1 3 13
mode_color 1 4 10
mode_color 1 5 3
mode_color 2 0 10
mode_color 2 1 11
mode_color 2 2 4
mode_color 2 3 13
mode_color 2 4 10
mode_color 2 5 3
mode_color 3 0 8
mode_color 3 1 11
mode_color 3 2 4
mode_color 3 3 13
mode_color 3 4 10
mode_color 3 5 3
mode_color 4 0 7
mode_color 4 1 11
mode_color 4 2 3
mode_color 4 3 13
mode_color 4 4 10
mode_color 4 5 3
mode_color 5 0 9
mode_color 5 1 11
mode_color 5 2 2
mode_color 5 3 13
mode_color 5 4 10
mode_color 5 5 3
mode_color 6 0 6
mode_color 6 1 10
mode_color 6 2 1
mode_color 6 3 0
mode_color 6 4 0
mode_color 6 5 2
mode_color 6 6 3
mode_color 6 7 6
mode_color 6 8 0
mode_color 6 9 0
mode_color 6 10 0

set looptime = 1000
set i2c_overclock = OFF
set gyro_sync = ON
set gyro_sync_denom = 1
set acc_task_frequency = 500
set attitude_task_frequency = 250
set async_mode = NONE
set mid_rc = 1500
set min_check = 1100
set max_check = 1900
set rssi_channel = 0
set rssi_scale = 30
set rssi_ppm_invert = OFF
set rc_smoothing = ON
set input_filtering_mode = OFF
set min_throttle = 1150
set max_throttle = 2000
set min_command = 1000
set 3d_deadband_low = 1406
set 3d_deadband_high = 1514
set 3d_neutral = 1460
set 3d_deadband_throttle = 1000
set motor_pwm_rate = 2000
set motor_pwm_protocol = ONESHOT125
set fixed_wing_auto_arm = OFF
set disarm_kill_switch = ON
set auto_disarm_delay = 10
set small_angle = 25
set reboot_character = 82
set gps_provider = UBLOX
set gps_sbas_mode = AUTO
set gps_dyn_model = AIR_4G
set gps_auto_config = ON
set gps_auto_baud = ON
set inav_auto_mag_decl = ON
set inav_accz_unarmedcal = ON
set inav_use_gps_velned = ON
set inav_gps_delay = 200
set inav_gps_min_sats = 6
set inav_w_z_baro_p =  0.350
set inav_w_z_gps_p =  0.200
set inav_w_z_gps_v =  0.500
set inav_w_xy_gps_p =  1.000
set inav_w_xy_gps_v =  2.000
set inav_w_z_res_v =  0.500
set inav_w_xy_res_v =  0.500
set inav_w_acc_bias =  0.010
set inav_max_eph_epv =  1000.000
set inav_baro_epv =  100.000
set nav_disarm_on_landing = OFF
set nav_use_midthr_for_althold = ON
set nav_extra_arming_safety = ON
set nav_user_control_mode = ATTI
set nav_position_timeout = 5
set nav_wp_radius = 100
set nav_max_speed = 300
set nav_max_climb_rate = 500
set nav_manual_speed = 500
set nav_manual_climb_rate = 500
set nav_landing_speed = 200
set nav_land_slowdown_minalt = 500
set nav_land_slowdown_maxalt = 2000
set nav_emerg_landing_speed = 500
set nav_min_rth_distance = 500
set nav_rth_climb_first = ON
set nav_rth_tail_first = OFF
set nav_rth_alt_mode = AT_LEAST
set nav_rth_altitude = 2000
set nav_mc_bank_angle = 30
set nav_mc_hover_thr = 1480
set nav_mc_auto_disarm_delay = 2000
set nav_fw_cruise_thr = 1400
set nav_fw_min_thr = 1200
set nav_fw_max_thr = 1700
set nav_fw_bank_angle = 20
set nav_fw_climb_angle = 20
set nav_fw_dive_angle = 15
set nav_fw_pitch2thr = 10
set nav_fw_roll2pitch = 75
set nav_fw_loiter_radius = 5000
set nav_fw_launch_velocity = 300
set nav_fw_launch_accel = 1863
set nav_fw_launch_detect_time = 40
set nav_fw_launch_thr = 1700
set nav_fw_launch_motor_delay = 500
set nav_fw_launch_timeout = 5000
set nav_fw_launch_climb_angle = 10
set serialrx_provider = SBUS
set spektrum_sat_bind = 0
set telemetry_switch = OFF
set telemetry_inversion = ON
set frsky_default_lattitude =  0.000
set frsky_default_longitude =  0.000
set frsky_coordinates_format = 0
set frsky_unit = IMPERIAL
set frsky_vfas_precision = 0
set frsky_vfas_cell_voltage = OFF
set hott_alarm_sound_interval = 5
set smartport_uart_unidir = OFF
set battery_capacity = 0
set vbat_scale = 110
set vbat_max_cell_voltage = 43
set vbat_min_cell_voltage = 32
set vbat_warning_cell_voltage = 35
set current_meter_scale = 285
set current_meter_offset = 0
set multiwii_current_meter_output = OFF
set current_meter_type = ADC
set align_gyro = DEFAULT
set align_acc = DEFAULT
set align_mag = CW270FLIP
set align_board_roll = 0
set align_board_pitch = -4
set align_board_yaw = 0
set gyro_lpf = 42HZ
set moron_threshold = 32
set imu_dcm_kp = 2500
set imu_dcm_ki = 50
set imu_dcm_kp_mag = 10000
set imu_dcm_ki_mag = 0
set pos_hold_deadband = 20
set alt_hold_deadband = 10
set yaw_motor_direction = 1
set yaw_jump_prevention_limit = 200
set tri_unarmed_servo = ON
set servo_lowpass_freq = 400
set servo_lowpass_enable = OFF
set servo_center_pulse = 1500
set servo_pwm_rate = 50
set failsafe_delay = 10
set failsafe_recovery_delay = 5
set failsafe_off_delay = 200
set failsafe_throttle = 1000
set failsafe_kill_switch = OFF
set failsafe_throttle_low_delay = 100
set failsafe_procedure = RTH
set rx_min_usec = 885
set rx_max_usec = 2115
set acc_hardware = MPU6050
set baro_use_median_filter = ON
set baro_hardware = MS5611
set mag_hardware = HMC5883
set blackbox_rate_num = 1
set blackbox_rate_denom = 1
set blackbox_device = SPIFLASH
set magzero_x = 12
set magzero_y = -126
set magzero_z = -117
set acczero_x = 32
set acczero_y = 17
set acczero_z = 61
set ledstrip_visual_beeper = OFF
set accgain_x = 4071
set accgain_y = 4090
set accgain_z = 4046

# rxfail
rxfail 0 a
rxfail 1 a
rxfail 2 a
rxfail 3 h
rxfail 4 s 1100
rxfail 5 s 1500
rxfail 6 s 1500
rxfail 7 h
rxfail 8 s 1100
rxfail 9 s 1900
rxfail 10 s 1500
rxfail 11 s 1500
rxfail 12 s 1500
rxfail 13 h
rxfail 14 h
rxfail 15 h
rxfail 16 h
rxfail 17 h

# dump profile

# profile
profile 0

# aux
aux 0 1 4 900 1300
aux 1 2 4 1300 1700
aux 2 19 5 1300 2100
aux 3 22 7 1700 2100
aux 4 6 7 900 1300
aux 5 3 5 1300 2100
aux 6 9 5 1700 2100
aux 7 8 0 1500 2100
aux 8 18 6 1700 2100
aux 9 20 6 900 1300
aux 10 0 0 900 900
aux 11 0 0 900 900
aux 12 0 0 900 900
aux 13 0 0 900 900
aux 14 0 0 900 900
aux 15 0 0 900 900
aux 16 0 0 900 900
aux 17 0 0 900 900
aux 18 0 0 900 900
aux 19 0 0 900 900

# adjrange
adjrange 0 0 0 900 900 0 0
adjrange 1 0 0 900 900 0 0
adjrange 2 0 0 900 900 0 0
adjrange 3 0 0 900 900 0 0
adjrange 4 0 0 900 900 0 0
adjrange 5 0 0 900 900 0 0
adjrange 6 0 0 900 900 0 0
adjrange 7 0 0 900 900 0 0
adjrange 8 0 0 900 900 0 0
adjrange 9 0 0 900 900 0 0
adjrange 10 0 0 900 900 0 0
adjrange 11 0 0 900 900 0 0

# rxrange
rxrange 0 1000 2000
rxrange 1 1000 2000
rxrange 2 1000 2000
rxrange 3 1000 2000

# servo
servo 0 1000 2000 1500 90 90 100 5
servo 1 1000 2000 1500 90 90 100 6
servo 2 1000 2000 1500 90 90 100 -1
servo 3 1000 2000 1500 90 90 100 -1
servo 4 1000 2000 1500 90 90 100 -1
servo 5 1000 2000 1500 90 90 100 -1
servo 6 1000 2000 1500 90 90 100 -1
servo 7 1000 2000 1500 90 90 100 -1

set nav_alt_p = 50
set nav_alt_i = 0
set nav_alt_d = 0
set nav_vel_p = 100
set nav_vel_i = 50
set nav_vel_d = 10
set nav_pos_p = 65
set nav_pos_i = 120
set nav_pos_d = 10
set nav_posr_p = 180
set nav_posr_i = 15
set nav_posr_d = 100
set nav_navr_p = 10
set nav_navr_i = 5
set nav_navr_d = 8
set deadband = 5
set yaw_deadband = 5
set throttle_tilt_comp_str = 0
set flaperon_throw_offset = 250
set flaperon_throw_inverted = OFF
set gimbal_mode = NORMAL
set fw_iterm_throw_limit = 165
set mode_range_logic_operator = OR
set default_rate_profile = 0
set mag_declination = 230
set mag_hold_rate_limit = 90
set p_pitch = 65
set i_pitch = 10
set d_pitch = 15
set p_roll = 40
set i_roll = 10
set d_roll = 15
set p_yaw = 65
set i_yaw = 10
set d_yaw = 15
set p_level = 20
set i_level = 15
set d_level = 75
set max_angle_inclination_rll = 500
set max_angle_inclination_pit = 500
set gyro_soft_lpf_hz = 60
set acc_soft_lpf_hz = 15
set dterm_lpf_hz = 40
set yaw_lpf_hz = 30
set yaw_p_limit = 300
set iterm_ignore_threshold = 200
set yaw_iterm_ignore_threshold = 50
set rate_accel_limit_roll_pitch = 0
set rate_accel_limit_yaw = 10000

# dump rates

# rateprofile
rateprofile 0

set rc_expo = 80
set rc_yaw_expo = 40
set thr_mid = 45
set thr_expo = 0
set roll_rate = 20
set pitch_rate = 20
set yaw_rate = 20
set tpa_rate = 0
set tpa_breakpoint = 1500

# 

Sorry, don't have bootlog at the moment, will post it at the weekend.

**Comment by gurkenfolie** [comment]
inav 1.21 servo output works like it should, the newer tested versions didnt work for me with the same settings (tested 1.3 and 1.5).

dump:
# dump

# version
# INAV/SPRACINGF3 1.2.1 Sep 19 2016 / 08:24:09 (36954a8)
# pflags
# Persistent config flags: 0x00000001
# dump master

# mixer
mixer QUADX
mmix reset
smix reset


# feature
feature -RX_PPM
feature -VBAT
feature -INFLIGHT_ACC_CAL
feature -RX_SERIAL
feature -MOTOR_STOP
feature -SERVO_TILT
feature -SOFTSERIAL
feature -GPS
feature -FAILSAFE
feature -SONAR
feature -TELEMETRY
feature -CURRENT_METER
feature -3D
feature -RX_PARALLEL_PWM
feature -RX_MSP
feature -RSSI_ADC
feature -LED_STRIP
feature -DISPLAY
feature -ONESHOT125
feature -BLACKBOX
feature -CHANNEL_FORWARDING
feature -TRANSPONDER
feature -AIRMODE
feature -SUPEREXPO
feature -VTX
feature -RX_NRF24
feature -SOFTSPI
feature RX_PPM
feature SERVO_TILT
feature GPS
feature FAILSAFE
feature ONESHOT125
feature BLACKBOX
feature CHANNEL_FORWARDING


# beeper
beeper GYRO_CALIBRATED
beeper RX_LOST
beeper RX_LOST_LANDING
beeper DISARMING
beeper ARMING
beeper ARMING_GPS_FIX
beeper BAT_CRIT_LOW
beeper BAT_LOW
beeper GPS_STATUS
beeper RX_SET
beeper ACC_CALIBRATION
beeper ACC_CALIBRATION_FAIL
beeper READY_BEEP
beeper MULTI_BEEPS
beeper DISARM_REPEAT
beeper ARMED
beeper SYSTEM_INIT
beeper ON_USB


# map
map AETR1234


# serial
serial 0 1 115200 38400 0 115200
serial 1 2 115200 57600 0 115200
serial 2 1 19200 38400 0 115200


# led
led 0 0,0::C:0
led 1 0,0::C:0
led 2 0,0::C:0
led 3 0,0::C:0
led 4 0,0::C:0
led 5 0,0::C:0
led 6 0,0::C:0
led 7 0,0::C:0
led 8 0,0::C:0
led 9 0,0::C:0
led 10 0,0::C:0
led 11 0,0::C:0
led 12 0,0::C:0
led 13 0,0::C:0
led 14 0,0::C:0
led 15 0,0::C:0
led 16 0,0::C:0
led 17 0,0::C:0
led 18 0,0::C:0
led 19 0,0::C:0
led 20 0,0::C:0
led 21 0,0::C:0
led 22 0,0::C:0
led 23 0,0::C:0
led 24 0,0::C:0
led 25 0,0::C:0
led 26 0,0::C:0
led 27 0,0::C:0
led 28 0,0::C:0
led 29 0,0::C:0
led 30 0,0::C:0
led 31 0,0::C:0


# color
color 0 0,0,0
color 1 0,255,255
color 2 0,0,255
color 3 30,0,255
color 4 60,0,255
color 5 90,0,255
color 6 120,0,255
color 7 150,0,255
color 8 180,0,255
color 9 210,0,255
color 10 240,0,255
color 11 270,0,255
color 12 300,0,255
color 13 330,0,255
color 14 0,0,0
color 15 0,0,0


# mode_color
mode_color 0 0 1
mode_color 0 1 11
mode_color 0 2 2
mode_color 0 3 13
mode_color 0 4 10
mode_color 0 5 3
mode_color 1 0 5
mode_color 1 1 11
mode_color 1 2 3
mode_color 1 3 13
mode_color 1 4 10
mode_color 1 5 3
mode_color 2 0 10
mode_color 2 1 11
mode_color 2 2 4
mode_color 2 3 13
mode_color 2 4 10
mode_color 2 5 3
mode_color 3 0 8
mode_color 3 1 11
mode_color 3 2 4
mode_color 3 3 13
mode_color 3 4 10
mode_color 3 5 3
mode_color 4 0 7
mode_color 4 1 11
mode_color 4 2 3
mode_color 4 3 13
mode_color 4 4 10
mode_color 4 5 3
mode_color 5 0 9
mode_color 5 1 11
mode_color 5 2 2
mode_color 5 3 13
mode_color 5 4 10
mode_color 5 5 3
mode_color 6 0 6
mode_color 6 1 10
mode_color 6 2 1
mode_color 6 3 0
mode_color 6 4 0
mode_color 6 5 2
mode_color 6 6 3
mode_color 6 7 6
mode_color 6 8 0
mode_color 6 9 0
mode_color 6 10 0

set looptime = 1000
set i2c_overclock = OFF
set gyro_sync = ON
set gyro_sync_denom = 8
set mid_rc = 1500
set min_check = 1020
set max_check = 1900
set rssi_channel = 0
set rssi_scale = 30
set rssi_ppm_invert = OFF
set rc_smoothing = ON
set input_filtering_mode = OFF
set min_throttle = 1025
set max_throttle = 1900
set min_command = 1000
set servo_center_pulse = 1500
set 3d_deadband_low = 1406
set 3d_deadband_high = 1514
set 3d_neutral = 1460
set 3d_deadband_throttle = 1000
set motor_pwm_rate = 400
set servo_pwm_rate = 50
set disarm_kill_switch = OFF
set auto_disarm_delay = 1
set small_angle = 25
set reboot_character = 82
set gps_provider = UBLOX
set gps_sbas_mode = NONE
set gps_dyn_model = AIR_1G
set gps_auto_config = ON
set gps_auto_baud = ON
set inav_auto_mag_decl = ON
set inav_accz_unarmedcal = ON
set inav_use_gps_velned = ON
set inav_gps_delay = 200
set inav_gps_min_sats = 6
set inav_w_z_baro_p =  0.350
set inav_w_z_gps_p =  0.200
set inav_w_z_gps_v =  0.500
set inav_w_xy_gps_p =  1.000
set inav_w_xy_gps_v =  2.000
set inav_w_z_res_v =  0.500
set inav_w_xy_res_v =  0.500
set inav_w_acc_bias =  0.010
set inav_max_eph_epv =  1000.000
set inav_baro_epv =  100.000
set nav_disarm_on_landing = ON
set nav_use_midthr_for_althold = OFF
set nav_extra_arming_safety = OFF
set nav_user_control_mode = ATTI
set nav_position_timeout = 5
set nav_wp_radius = 100
set nav_max_speed = 500
set nav_max_climb_rate = 500
set nav_manual_speed = 500
set nav_manual_climb_rate = 500
set nav_landing_speed = 200
set nav_land_slowdown_minalt = 380
set nav_land_slowdown_maxalt = 800
set nav_emerg_landing_speed = 500
set nav_min_rth_distance = 200
set nav_rth_tail_first = OFF
set nav_rth_alt_mode = AT_LEAST
set nav_rth_altitude = 700
set nav_mc_bank_angle = 40
set nav_mc_hover_thr = 1440
set nav_mc_auto_disarm_delay = 200
set nav_fw_cruise_thr = 1400
set nav_fw_min_thr = 1200
set nav_fw_max_thr = 1700
set nav_fw_bank_angle = 20
set nav_fw_climb_angle = 20
set nav_fw_dive_angle = 15
set nav_fw_pitch2thr = 10
set nav_fw_roll2pitch = 75
set nav_fw_loiter_radius = 5000
set serialrx_provider = SPEK1024
set spektrum_sat_bind = 0
set telemetry_switch = OFF
set telemetry_inversion = ON
set frsky_default_lattitude =  0.000
set frsky_default_longitude =  0.000
set frsky_coordinates_format = 0
set frsky_unit = IMPERIAL
set frsky_vfas_precision = 0
set hott_alarm_sound_interval = 5
set battery_capacity = 0
set vbat_scale = 110
set vbat_max_cell_voltage = 43
set vbat_min_cell_voltage = 33
set vbat_warning_cell_voltage = 35
set current_meter_scale = 400
set current_meter_offset = 0
set multiwii_current_meter_output = OFF
set current_meter_type = ADC
set align_gyro = DEFAULT
set align_acc = DEFAULT
set align_mag = DEFAULT
set align_board_roll = 1800
set align_board_pitch = 0
set align_board_yaw = 2700
set gyro_lpf = 256HZ
set moron_threshold = 32
set imu_dcm_kp = 2500
set imu_dcm_ki = 50
set imu_dcm_kp_mag = 10000
set imu_dcm_ki_mag = 0
set pos_hold_deadband = 50
set alt_hold_deadband = 100
set yaw_motor_direction = 1
set yaw_jump_prevention_limit = 200
set tri_unarmed_servo = ON
set servo_lowpass_freq = 400
set servo_lowpass_enable = OFF
set failsafe_delay = 10
set failsafe_off_delay = 200
set failsafe_throttle = 1300
set failsafe_kill_switch = OFF
set failsafe_throttle_low_delay = 100
set failsafe_procedure = SET-THR
set rx_min_usec = 885
set rx_max_usec = 2115
set acc_hardware = 0
set baro_use_median_filter = ON
set baro_hardware = 0
set mag_hardware = 0
set blackbox_rate_num = 1
set blackbox_rate_denom = 3
set blackbox_device = SERIAL
set magzero_x = 0
set magzero_y = -25
set magzero_z = 17
set acczero_x = 85
set acczero_y = -8
set acczero_z = 176
set ledstrip_visual_beeper = OFF
set accgain_x = 4107
set accgain_y = 4062
set accgain_z = 4041

# rxfail
rxfail 0 a
rxfail 1 a
rxfail 2 a
rxfail 3 h
rxfail 4 h
rxfail 5 h
rxfail 6 h
rxfail 7 h
rxfail 8 h
rxfail 9 h
rxfail 10 h
rxfail 11 h
rxfail 12 h
rxfail 13 h
rxfail 14 h
rxfail 15 h
rxfail 16 h
rxfail 17 h

# dump profile

# profile
profile 0

# aux
aux 0 1 3 1300 1700
aux 1 2 3 900 1225
aux 2 5 4 1300 1700
aux 3 3 2 1700 1925
aux 4 3 2 1225 1325
aux 5 10 2 1225 1350
aux 6 9 2 1950 2025
aux 7 9 2 900 975
aux 8 24 4 1700 2100
aux 9 27 4 1725 2100
aux 10 0 0 900 900
aux 11 0 0 900 900
aux 12 0 0 900 900
aux 13 0 0 900 900
aux 14 0 0 900 900
aux 15 0 0 900 900
aux 16 0 0 900 900
aux 17 0 0 900 900
aux 18 0 0 900 900
aux 19 0 0 900 900

# adjrange
adjrange 0 0 0 900 900 0 0
adjrange 1 0 0 900 900 0 0
adjrange 2 0 0 900 900 0 0
adjrange 3 0 0 900 900 0 0
adjrange 4 0 0 900 900 0 0
adjrange 5 0 0 900 900 0 0
adjrange 6 0 0 900 900 0 0
adjrange 7 0 0 900 900 0 0
adjrange 8 0 0 900 900 0 0
adjrange 9 0 0 900 900 0 0
adjrange 10 0 0 900 900 0 0
adjrange 11 0 0 900 900 0 0

# rxrange
rxrange 0 1000 2000
rxrange 1 1000 2000
rxrange 2 1000 2000
rxrange 3 1000 2000

# servo
servo 0 1000 2000 1500 90 90 100 4
servo 1 1000 2000 1500 90 90 100 5
servo 2 1000 2000 1500 90 90 100 -1
servo 3 1000 2000 1500 90 90 100 -1
servo 4 1000 2000 1500 90 90 100 -1
servo 5 1000 2000 1500 90 90 100 -1
servo 6 1000 2000 1500 90 90 100 -1
servo 7 1000 2000 1500 90 90 100 -1

set nav_alt_p = 50
set nav_alt_i = 0
set nav_alt_d = 0
set nav_vel_p = 100
set nav_vel_i = 50
set nav_vel_d = 10
set nav_pos_p = 65
set nav_pos_i = 120
set nav_pos_d = 10
set nav_posr_p = 180
set nav_posr_i = 15
set nav_posr_d = 100
set nav_navr_p = 10
set nav_navr_i = 5
set nav_navr_d = 8
set deadband = 10
set yaw_deadband = 15
set throttle_tilt_comp_str = 0
set flaperon_throw_offset = 250
set flaperon_throw_inverted = OFF
set mode_range_logic_operator = OR
set default_rate_profile = 0
set gimbal_mode = NORMAL
set mag_declination = 330
set mag_hold_rate_limit = 90
set p_pitch = 70
set i_pitch = 56
set d_pitch = 66
set p_roll = 65
set i_roll = 56
set d_roll = 66
set p_yaw = 61
set i_yaw = 20
set d_yaw = 0
set p_level = 15
set i_level = 15
set d_level = 10
set max_angle_inclination_rll = 450
set max_angle_inclination_pit = 450
set gyro_soft_lpf_hz = 60
set acc_soft_lpf_hz = 15
set dterm_lpf_hz = 40
set yaw_lpf_hz = 30
set yaw_p_limit = 300
set iterm_ignore_threshold = 200
set yaw_iterm_ignore_threshold = 50
set rate_accel_limit_roll_pitch = 0
set rate_accel_limit_yaw = 10000

# dump rates

# rateprofile
rateprofile 0

set rc_expo = 60
set rc_yaw_expo = 20
set thr_mid = 50
set thr_expo = 0
set roll_rate = 120
set pitch_rate = 100
set yaw_rate = 40
set tpa_rate = 0
set tpa_breakpoint = 1500

#

**Comment by oleost** [comment]
Please provide dump of iNav 1.5 . Also remember you need to enable pwm output on latest iNav.

And also share "bootlog" dump from iNav 1.5

**Comment by jflyper** [comment]
@Xe54 
- iNav 1.5.1 SPRACINGF3 Gimbal servo works as expected.
<img width="805" alt="2017-01-06 22 32 24" src="https://cloud.githubusercontent.com/assets/14850998/21719369/bd091804-d460-11e6-8846-274e2b56658b.png">

- You have different AUX channels assigned in your dumps (5 & 6 v.s. 4 & 5)

- Don't turn on channel forwarding. That does something totally different.

- bootlog dump is preferred.

**Comment by Xe54** [comment]
OK, I spent a half of the weekent to figure out what the problem is. And I must confess, I still have no idea. Generally it seems to work, but not with my setup. 
My situation is:
1. My setup works properly with any version of Cleanflight
2. My setup doesn't work with iNav regardless of the version
3. A simplier iNav setup works with my 3D feiyu tech gimbal
4. Non of this iNav setups works with my 2D feiyu tech gimbal

I will need more time to figure out the reason. I know that my 2D gimbal is pretty sensitive to the input signals at start up. Could it be another behavior/timing of iNav compared to Cleanflight at start up, that constitutes a problem for my gimbal? Anyway here is the bootlog, I hope it helps somehow:

# bootlog
Time Evt            Description  Parameters
   0:  0          CONFIG_LOADED 
   0:  1       SYSTEM_INIT_DONE 
 501: 19   TIMER_CHANNEL_MAPPED  (1, 1, 0, 2)
 501: 19   TIMER_CHANNEL_MAPPED  (2, 2, 0, 2)
 501: 19   TIMER_CHANNEL_MAPPED  (3, 3, 0, 2)
 501: 19   TIMER_CHANNEL_MAPPED  (4, 4, 0, 2)
 501: 19   TIMER_CHANNEL_MAPPED  (5, 5, 0, 2)
 501: 19   TIMER_CHANNEL_MAPPED  (6, 6, 0, 2)
 501: 19   TIMER_CHANNEL_MAPPED  (7, 6, 1, 3)
 501: 19   TIMER_CHANNEL_MAPPED  (8, 6, 2, 3)
 501: 18  TIMER_CHANNEL_SKIPPED  (9, 6, 2, 3)
 501: 18  TIMER_CHANNEL_SKIPPED  (10, 6, 2, 3)
 501: 18  TIMER_CHANNEL_SKIPPED  (11, 6, 2, 3)
 501: 18  TIMER_CHANNEL_SKIPPED  (12, 6, 2, 3)
 501:  2          PWM_INIT_DONE 
 511:  3       EXTRA_BOOT_DELAY 
1048:  9         GYRO_DETECTION  (2, 0, 0, 0)
1163: 10          ACC_DETECTION  (3, 0, 0, 0)
1177: 11         BARO_DETECTION  (3, 0, 0, 0)
1178: 12          MAG_DETECTION  (2, 0, 0, 0)
2837: 13  RANGEFINDER_DETECTION  (0, 0, 0, 0)
2837:  4       SENSOR_INIT_DONE 
3388:  5          GPS_INIT_DONE 
3438:  7    TELEMETRY_INIT_DONE 
3489:  8           SYSTEM_READY 

#

**Comment by oleost** [comment]
Okey, now I'm testing out `Servo Gimbal` feature

iNav servo tilt working on both pin 7 and pin 8 ( spracingf3 )
CLI: http://pastebin.com/V5raTg9x 
Bootlog: http://pastebin.com/GEn7LuvZ

I see my bootlog is different from yours. @digitalentity Do you know why our bootlog is different in regards to the first mapped timer?

**Comment by oleost** [comment]
I started working on a Wiki page to assist in using Servo gimbal feature.

https://github.com/iNavFlight/inav/wiki/Connecting-a-Gimbal-to-iNav

**Comment by Xe54** [comment]
Thanks, nice Wiki page. Can you please also explain, what live mode makes?

**Comment by oleost** [comment]
Live mode is simple that the change you make in that tab is "live".

You dont need to press save to test example new midpoint, just enter new midpoint and it set right away instead of needing to press save.

**Comment by Xe54** [comment]
Ok, I think I found something. As soon as I enable Serial RX for UART1 my gimbal stops to work. Can you check this please? I'm using UART1 for SBUS communication with FC.

**Comment by oleost** [comment]
UART1? That's shared with USB?

**Comment by Xe54** [comment]
Forget all I said above. I found the reason and that's not iNav. The problem was caused by BEC that I'm using. It seems to provide some interferences/noises over power supply, that are confusing my gimbal. For the reason I can't explain it works fine with CF. Anyway exchanging of BEC solved the problem. Here is the peace of hardware that I used as BEC
![dsc_0460](https://cloud.githubusercontent.com/assets/18517887/21963172/d1812d68-db35-11e6-95da-4bea8b1af613.JPG)
The bug is closed.
